ecospheres_universe/feed_universe.py

```diff
@@ -182,61 +163,13 @@ def delete_topic_datasets(self, topic, datasets):
                     raise
                 verbose(e)
 
-    #FIXME: doesn't work on huge topics
     @elapsed
-    def delete_all_topic_datasets(self, topic):
-        # workaround for missing "delete all" endpoint
-        # 1. get needed info from topic
-        url = f"{self.base_url}/api/2/topics/{topic}/datasets/?page_size=1"
-        r = session.get(url, headers={'X-Fields': 'data{id}'})
-        r.raise_for_status()
-        c = r.json()
-        single = [d['id'] for d in c.get('data', [])] # page_size=1
-
+    def delete_all_topic_elements(self, topic):
         url = f"{self.base_url}/api/2/topics/{topic}/"
-        r = session.get(url, headers={'X-Fields': 'tags'})
-        r.raise_for_status()
-        c = r.json()
-        tags = c.get('tags', [])
-
-        # 2. override (api v1) existing datasets with list of 1 element
-        url = f"{self.base_url}/api/1/topics/{topic}/"
-        data = json.dumps({'datasets': single, 'tags': tags})
-        headers = {'Content-Type': 'application/json', 'X-API-KEY': self.token}
         if not self.dry_run:
-            session.put(url, data=data, headers=headers).raise_for_status()
-
-        # 3. delete element from step 2
-        if not self.dry_run:
-            self.delete_topic_datasets(topic, single)
-
-    #FIXME: doesn't work on huge topics
-    @elapsed
-    def slow_delete_all_topic_datasets(self, topic):
-        print("Delete v2")
-        datasets = self.get_topic_datasets(topic)
-        self.delete_topic_datasets(topic, datasets)
-        print("Delete hack")
-        try:
-            api.delete_all_topic_datasets(topic)
-        except:
-            pass
-        print("Delete v1")
-        datasets = self.get_topic_datasets_v1(topic)
-        self.delete_topic_datasets(topic, datasets)
-
-
-def check_sync(topic_slug: str, org: Organization, datasets: list):
-    topic_id = api.get_topic_id(topic_slug)
-    topic_datasets = api.get_topic_datasets(topic_slug, organization_id=org.id)
-    datasets_search_count = api.search_datasets_count(topic_id, organization_id=org.id)
-    if not(len(topic_datasets) == len(datasets) == datasets_search_count):
-        print(f"Datasets for '{org.slug}' are NOT in sync", file=sys.stderr)
-        print(f"  - topic datasets : {len(topic_datasets)}", file=sys.stderr)
-        print(f"  - universe       : {len(datasets)}", file=sys.stderr)
-        print(f"  - search datasets: {datasets_search_count}", file=sys.stderr)
-    else:
-        verbose(f"Datasets for '{org.slug}' are in sync ({len(datasets)} datasets)")
+            headers = {'Content-Type': 'application/json', 'X-API-KEY': self.token}
+            # FIXME: test this
```

Comment:
Were there changes in the backend that will make this work now?

Reply:
I forgot to test that, and it does not work. After a loooong hunt, I stumbled on this good old bug https://github.com/kvesteri/wtforms-json/issues/43 😬 It explains why it has never worked and won't work without upstream changes. I'll try to circumvent.

Reply:
https://github.com/opendatateam/udata/pull/3318/commits/7f47e03b0b79e04553ae09dc32da170fcccf010b